### PR TITLE
6 view an individual review

### PR DESCRIPTION
**Pull request by Proto-Will**

### CODE REVIEW

File: src/pages/AllReviews.jsx [hunk 1]
```diff
@@ -13,12 +13,11 @@ export default function AllReviews() {
...
                 setIsLoading(false);
             })
     }, [])
-
-if (isLoading) return <p>Loading...</p>
+    
+    if(isLoading) return <p>Loading...</p>
 
 return (
         <section>
```
Comment (Poonam-raj) [inline src/pages/AllReviews.jsx:23]: in line 28 of this component you need a key, currently getting this error in the browser console when I look at the list of reviews:
image

Comment (Poonam-raj) [inline src/pages/AllReviews.jsx:23]: Sidenote on the all reviews component, the styling in mobile view is a bit less enjoyable as it cuts off and is a sideways scroll to see all the reviews:
image

might be worth addressing mobile view either in this branch or a future branch

File: src/pages/AllCategories.jsx [hunk 2]
```diff
@@ -15,8 +15,8 @@ export default function AllCategories() {
                 setIsLoading(false);
             })
     }, [])
-
-if (isLoading) return <p>Loading...</p>
+    
+    if(isLoading) return <p>Loading...</p>
 
     return (
         <section>
```
Comment (Poonam-raj) [inline src/pages/AllCategories.jsx:22]: Getting the same react warning for viewing all categories:
image

make sure you have your 'key' props in that map

File: src/pages/CategorisedReviews.jsx [hunk 3]
```diff
@@ -6,16 +6,20 @@ import ReviewCard from "../components/ReviewCard";
...
             .then((reviews) => {
                 setReviewList(reviews.reviews);
+                setIsLoading(false)
             })
     }, [])
 
     let filtered = reviewList.filter((review) => {return review.category === category})
+    if(isLoading) return <p>Loading...</p>
 
     return (
```
Comment (Poonam-raj) [inline src/pages/CategorisedReviews.jsx:24]: Same key props warning happening for this component. Need to include key in the map

File: src/pages/IndividualReview.jsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+    return  (
+        <div>            
+            <ReviewCard review={review} />
+            <div>
+                <p>{review.review_body}</p>
+            </div>
+            <div>
+            <button className="buttons" onClick={() => navigate(`/reviews/${ID}/comments`)}>Comments</button>
+            </div>
```
Comment (Poonam-raj) [inline src/pages/IndividualReview.jsx]: Are these p and button elements wrapped in divs for styling? Might be better to let the CSS space them out and take these divs out

**Review by Poonam-raj — APPROVED** [review]
Few things worth addressing: 
- mobile view for any pages rendering lists of items
- your react warnings (missing key prop in mapped items)

The actual single review view is wonderful and works well.
The mobile view is not massively important for this ticket but please fix the missing 'key' props and feel free to merge when you're done

File: src/pages/IndividualReview.jsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        useEffect(() => {
+        setIsLoading(true)
+        fetch(`https://final-boardgame-api.herokuapp.com/api/reviews/${review_id}`)
+            .then((response) => response.json())
+            .then((data) => {
+                setReview(data.review);
+                setIsLoading(false)
+            })
+    }, [])
+    const ID = review_id
```
Comment (k-huggs) [inline src/pages/IndividualReview.jsx:23]: you already have access to review_id so you don't need to assign it to a variable

**Comment by k-huggs** [comment]
All good to merge!

**Pull request merged**